cmd/status.go

```diff
@@ -0,0 +1,197 @@
+					return err
+				}
+				if status.Role, err = getRole(status.Pid); err != nil {
+					return err
+				}
+			}
+
+			status.output = output
+			status.String()
+			return nil
+		},
+	}
+)
+
+func init() {
+	status = &K0sStatus{}
+	statusCmd.PersistentFlags().StringVarP(&output, "out", "o", "", "sets type of out put to json or yaml")
+}
+
+type K0sStatus struct {
+	Version  string
+	Pid      int
+	PPid     int
+	Role     string
+	SysInit  string
+	StubFile string
+	output   string
+}
+
+func (s K0sStatus) String() {
+	s.Version = build.Version
+	switch s.output {
+	case "json":
+		jsn, _ := json.MarshalIndent(s, "", "   ")
+		fmt.Println(string(jsn))
+	case "yaml":
+		ym, _ := yaml.Marshal(s)
+		fmt.Println(string(ym))
+	default:
+		fmt.Println("Version:", s.Version)
+		if s.Pid == 0 {
+			fmt.Println("K0s not running")
+		} else {
+			fmt.Println("Process ID:", s.Pid)
+			fmt.Println("Parent Process ID:", s.PPid)
+			fmt.Println("Role:", s.Role)
+		}
+
+		if s.SysInit != "" {
+			fmt.Println("Init System:", s.SysInit)
+
+		}
+		if s.StubFile != "" {
+			fmt.Println("Service file:", s.StubFile)
+		}
+	}
+
+}
+func getSysInit(role string) (sysInit string, stubFile string, err error) {
+	if role == "server+worker" {
+		role = "server"
+	}
+	if sysInit, err = install.GetSysInit(); err != nil {
+		return sysInit, stubFile, err
+	}
+	if sysInit == "linux-systemd" {
+		stubFile = fmt.Sprintf("/etc/systemd/system/k0s%s.service", role)
+		if _, err := os.Stat(stubFile); err != nil {
+			stubFile = ""
+		}
+	} else if sysInit == "linux-openrc" {
+		stubFile = fmt.Sprintf("/etc/init.d/k0s%s", role)
+		if _, err := os.Stat(stubFile); err != nil {
+			stubFile = ""
+		}
+	}
+
+	return sysInit, stubFile, err
+
+}
+
+func getRole(pid int) (role string, err error) {
+	if runtime.GOOS == "windows" {
+		return "worker", nil
+	}
+
+	var raw []byte
+	if raw, err = ioutil.ReadFile(fmt.Sprintf("/proc/%d/cmdline", pid)); err != nil {
+		return "", err
+	}
+	cmdln := string(raw)
+	if strings.Contains(cmdln, "enable-worker") {
+		return "server+worker", nil
+	} else if strings.Contains(cmdln, "server") {
+		return "server", nil
+	} else if strings.Contains(cmdln, "worker") {
+		return "worker", nil
+	}
+	return "", fmt.Errorf("k0s role is not found")
+}
+
+func getPid() (status *K0sStatus, err error) {
+	processList, err := ps.Processes()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, p := range processList {
+		if p.Executable() == "k0s" && hasChildren(p.Pid(), processList) {
+			status = &K0sStatus{Pid: p.Pid(),
+				PPid: p.PPid()}
+
+			return status, nil
+		}
+	}
+
+	return &K0sStatus{}, nil
+}
+
+func hasChildren(pid int, processes []ps.Process) bool {
+	for _, p := range processes {
+		if p.PPid() == pid {
+			return true
+		}
+	}
+	return false
+}
+
+func getProcessOwner(pid int) (string, error) {
+	stdout, err := exec.Command("ps", "-o", "user=", "-p", strconv.Itoa(pid)).Output()
```

Comment:
I think you could just use the [process.Username](https://pkg.go.dev/github.com/shirou/gopsutil/process#Process.Username)